### DISCUSSION TITLE
`retries` of `AsyncHTTPTransport` and `HTTPTransport` is not passes to `pools` with proxy but only ordinary `pool`

### DISCUSSION THREAD

**Discussion by NewUserHa** [encode/httpx#2988] [Potential Issue]
https://github.com/encode/httpx/blob/5b5f6d8e172910371b5901f6e21717097274c513/httpx/_transports/default.py#L258-L330
eg.
`AsyncConnectionPool` has the argument of retries passed in but `AsyncHTTPProxy` and `AsyncSOCKSProxy` have not.

Can I open a PR?

**Comment by karpetrosyan** [comment]
It appears that the `local_address` and `uds` arguments are also missing.

> Can I open a PR?

That would be great.

**Reply by lovelydinosaur** [reply]
The associated PRs are a little confusing to me...

Can we slow things down a little and deal with one aspect at a time?

> `AsyncConnectionPool` has the argument of retries passed in but `AsyncHTTPProxy` and `AsyncSOCKSProxy` have not.

I can see that `httpx.HTTPTransport` and `httpx.AsyncHTTPTransport` are passing `retries` in the connection pool case, but not in the proxy case. `httpcore` supports the argument in both case. I'd assume we should *just* start with a pull request resolving this?

**Comment by NewUserHa** [comment]
PR: https://github.com/encode/httpx/pull/2997
